src/liballoc/string.rs

```diff
@@ -2206,13 +2206,40 @@ impl<'a> From<&'a str> for String {
 #[cfg(not(test))]
 #[stable(feature = "string_from_box", since = "1.18.0")]
 impl From<Box<str>> for String {
+    /// Converts the given boxed `str` slice to a `String`.
+    /// It is notable that the `str` slice is owned.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// let s1 : String = String::from("hello world");
+    /// let s2 : Box<str> = s1.into_boxed_str();
+    /// let s3 : String = String::from(s2);
```

Comment:
Code style:
``let s1 : String`` should be ``let s1: String``
Same goes to other let statements with annotations.